Update Prompt API spec and explainer with Sampling Params changes (#206)
explainers-by-googlers/prompt-api: README.md

*   `"predictable"`: For focused outputs with minimal variation.
*   `"balanced"` (default): The standard preset for most conversational interactions.
*   `"creative"`: For tasks where variety and creativity are preferred over strict factual reproducibility.
*   `"most-creative"`: For maximum diversity of tokens and creative brainstorming.

Example:
```js
const creativeSession = await LanguageModel.create({
  samplingMode: "creative"
});
console.log(creativeSession.samplingMode); // "creative"
```

The resolved `samplingMode` used to create the session is exposed as a read-only attribute on the session object.

### Legacy: Configuration of per-session raw parameters

**Deprecation Notice:** The `topK` and `temperature` options for `LanguageModel.create()`, the `LanguageModel.params()` static method, and the `languageModel.topK` and `languageModel.temperature` instance attributes are now **deprecated**. These features are only functional within web extension contexts and will be ignored in standard web page contexts. They may be completely removed in a future release.

> To avoid breaking existing pages, standard web page contexts can still pass `topK` and `temperature` in the options object without throwing an error (to log deprecation warnings in the console), but they are ignored at runtime and the corresponding properties on the session object will be `undefined` (or fallback to default values).

Reviewer: ```suggestion
To avoid breaking existing pages, standard web page contexts can still pass `topK` and `temperature` in the options object without throwing an error (a deprecation warning will be logged in the console), but they are ignored at runtime and the corresponding properties on the session object will be `undefined` (or fallback to default values).
```

> Furthermore, in contexts where raw parameters are supported (e.g. Chrome Extensions), passing both `samplingMode` and a raw parameter (`topK` or `temperature`) will reject the `create()` promise with a `TypeError`.

Reviewer: ```suggestion
Furthermore, in contexts where raw parameters are supported (e.g. Web Extensions), passing both `samplingMode` and a raw parameter (`topK` or `temperature`) will reject the `create()` promise with a `TypeError`.
```

Note that `append()` can also cause [overflow](#tokenization-context-window-length-limits-and-overflow), in which case it will evict the oldest non-system prompts from the session and fire the `"contextoverflow"` event.

### Configuration of sampling modes

> For standard web page contexts, developers can specify a high-level `samplingMode` during session creation to configure the model's output variety and creativity without worrying about model-internal scalar parameters.

Reviewer 2: Drop the "For standard web page contexts" since this is experimentally available for web and extensions.
```suggestion
Developers can specify a high-level `samplingMode` during session creation to configure the model's output variety and creativity without worrying about model-internal scalar parameters.
```

The allowed values for `samplingMode` are:

> *   `"most-predictable"`: For tasks requiring strict consistency and reproducibility (e.g., code generation or factual extraction).

Reviewer 2: nit: mention testing, replace factual with content.

```suggestion
*   `"most-predictable"`: For tasks requiring strict consistency and reproducibility (e.g., testing, code generation, or content extraction).
```

*   `"predictable"`: For focused outputs with minimal variation.
*   `"balanced"` (default): The standard preset for most conversational interactions.

> *   `"creative"`: For tasks where variety and creativity are preferred over strict factual reproducibility.

Reviewer 2: ```suggestion
*   `"creative"`: For tasks where variety and creativity are preferred over strict reproducibility.
```

> *   `"most-creative"`: For maximum diversity of tokens and creative brainstorming.

Reviewer 2: ```suggestion
*   `"most-creative"`: For maximum diversity of output and creative brainstorming.
```

To avoid breaking existing pages, standard web page contexts can still pass `topK` and `temperature` in the options object without throwing an error (a deprecation warning will be logged in the console), but they are ignored at runtime and the corresponding properties on the session object will be `undefined` (or fallback to default values).

Furthermore, in contexts where raw parameters are supported (e.g. Web Extensions), passing both `samplingMode` and a raw parameter (`topK` or `temperature`) will reject the `create()` promise with a `TypeError`.

The `LanguageModel.params()` API, only available in extensions, can be used to query the default and maximum values for these parameters.

_The limited applicability and non-universal nature of these sampling hyperparameters are discussed further in [issue #42](https://github.com/webmachinelearning/prompt-api/issues/42) and [issue #203](https://github.com/webmachinelearning/prompt-api/issues/203)._

```js
// The topK and temperature members of the options object are deprecated. They will only be considered when

> // LanguageModel.create() is called from within a Chrome Extension. In web page contexts, they are ignored.

Reviewer: ```suggestion
// LanguageModel.create() is called from within a web extension. In web page contexts, they are ignored.
```

> // LanguageModel.create() is called from within a Chrome Extension. In web page contexts, they are ignored.

Reviewer 2: ```suggestion
// LanguageModel.create() is called from within a web extension. In web page contexts, they are ignored.
```

const customSession = await LanguageModel.create({
  temperature: 0.8,
  topK: 10
});
// This interface and all its attributes (`defaultTopK`, `maxTopK`, `defaultTemperature`, `maxTemperature`)

> // are now only available within Chrome Extension contexts. Web pages can no longer call this method.

Reviewer: ```suggestion
// are now only available within web extension contexts. Web pages can no longer call this method.
```

> // are now only available within Chrome Extension contexts. Web pages can no longer call this method.

Reviewer 2: ```suggestion
// are now only available within web extension contexts. Web pages can no longer call this method.
```

If the language model is not available at all in this browser, `params()` will fulfill with `null`.

> Error-handling behavior (only applicable in contexts where legacy parameters are active, e.g. Chrome Extensions):

Reviewer: ```suggestion
Error-handling behavior (only applicable in contexts where legacy parameters are active, e.g. Web Extensions):
```
